### DISCUSSION TITLE
How to use hooks like `render:route`

### DISCUSSION THREAD

**Discussion by feerglas** [nuxt/nuxt#18023] [Questions]
In nuxt2, I was able to use the `render:route` hook like so:
```
this.nuxt.hook('render:route', (url, page) => {
 console.log('hook -> render:route');
});
```

If i try to do it in nuxt3 in my `nuxt.config.ts`, it doesn't work. What am I missing here?
```
export default defineNuxtConfig({
  hooks: {
    'render:route': () => {
       console.log('hook -> render:route');
    }
  },
}
```

I did not find anything about the issue in the nuxt3 docu. Any help appreciated :)

**Comment by technicallypete** [comment]
Yea, because for some reason there is 0 documentation on what all the "hooks" are..

**Reply by maxdzin** [reply]
@feerglas It seems that `render:route` hook does not work but I found a workaround by using `page:start` and `page:finish` hooks:

```
const nuxtApp = useNuxtApp()
nuxtApp.hook('page:start', () => {
  console.log('start')
  ...
})

nuxtApp.hook('page:finish', () => {
  console.log('finish')
  window.scrollTo(0, 0)
  ...
})
```
Maybe that will help you.
